ocaml/xapi/dbsync_master.ml

```diff
@@ -287,6 +289,42 @@ let ensure_vm_metrics_records_exist_noexn __context =
   Helpers.log_exn_continue "ensuring VM_metrics flags exist"
     ensure_vm_metrics_records_exist __context
 
+let setup_telemetry ~__context =
+  let pool = Helpers.get_pool ~__context in
+  let ref = Db.Pool.get_telemetry_uuid ~__context ~self:pool in
+  if ref = Ref.null then
+    Helpers.log_exn_continue "Setting up telemetry"
+      (fun () ->
+        Helpers.call_api_functions ~__context (fun rpc session_id ->
+            Client.Pool.reset_telemetry_uuid ~rpc ~session_id ~self:pool
+        ) ;
+        (* An exception will result in leaving the next schedule as default *)
+        let bound_hour =
+          match Db.Pool.get_telemetry_interval ~__context ~self:pool with
+          | `daily ->
+              1 * 24
+          | `weekly ->
+              7 * 24
+          | `monthly ->
+              30 * 24
+        in
+        let value =
+          let open Ptime in
+          (* A quiescent period (1 day) plus a random hour within a telemetry interval *)
+          Span.of_int_s (3600 * 24)
+          |> Span.add (Span.of_int_s (Random.int (bound_hour * 3600)))
+          |> add_span (Ptime_clock.now ())
+          |> Option.get
+          |> to_float_s
+          |> Xapi_stdext_date.Date.of_float
```

Comment:
```suggestion
          |> Xapi_stdext_date.Date.of_ptime
```

Reply:
Thanks! I forgot to `git pull` in the local repository hence didn't get the latest changes in `stdext`.